### PR TITLE
Add some docs to the doctrine/coding-standard project.

### PR DESCRIPTION
**Pull request by jwage**
Related to https://github.com/doctrine/doctrine-website/pull/176

Should I change the README.md to look like the other projects and just point to the docs with a few links? That way we don't have to maintain 2 sets of the same information about the project.

### CODE REVIEW

**Comment by alcaeus** [comment]
> Should I change the README.md to look like the other projects

Yes please, it makes more sense to use proper documentation than just a readme.

File: docs/en/reference/index.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+Installation
+============
+
+You have two possibilities to use the Doctrine Coding Standard with PHP_CodeSniffer in a particular project.
+
+Composer Project Dependency
+---------------------------
+
+You can install the Doctrine Coding Standard as a composer dependency to your particular project.
+Just add the following block to your project's ``composer.json`` file:
```
Comment (greg0ire) [inline docs/en/reference/index.rst]: That's not a block, that's a command, and you don't want to add it in this file :D

**Comment by jwage** [comment]
@alcaeus updated readme

File: README.md [hunk 2]
```diff
@@ -1,103 +1,11 @@
-Doctrine Coding Standard
-========================
+# Doctrine Coding Standard
 
-[![Build Status](https://img.shields.io/travis/doctrine/coding-standard/master.svg?style=flat-square)](http://travis-ci.org/doctrine/coding-standard)
-[![Total Downloads](https://img.shields.io/packagist/dt/doctrine/coding-standard.svg?style=flat-square)](https://packagist.org/packages/doctrine/coding-standard)
-[![Latest Stable Version](https://img.shields.io/packagist/v/doctrine/coding-standard.svg?style=flat-square)](https://packagist.org/packages/doctrine/coding-standard)
```
Comment (Majkl578) [inline README.md]: Please keep these badges.

File: README.md [hunk 3]
```diff
@@ -1,103 +1,13 @@
-Doctrine Coding Standard
-========================
+# Doctrine Coding Standard
 
 [![Build Status](https://img.shields.io/travis/doctrine/coding-standard/master.svg?style=flat-square)](http://travis-ci.org/doctrine/coding-standard)
 [![Total Downloads](https://img.shields.io/packagist/dt/doctrine/coding-standard.svg?style=flat-square)](https://packagist.org/packages/doctrine/coding-standard)
 [![Latest Stable Version](https://img.shields.io/packagist/v/doctrine/coding-standard.svg?style=flat-square)](https://packagist.org/packages/doctrine/coding-standard)
 
+The Doctrine Coding Standard is a set of PHPCS rules applied to all Doctrine projects.
```
Comment (morozov) [inline README.md]: I think the full name with the link was better than abbreviation without the link.

File: docs/en/reference/index.rst [hunk 4]
```diff
@@ -0,0 +1,94 @@
+Introduction
+============
+
+The Doctrine Coding Standard is a set of PHPCS rules applied to all Doctrine projects. It is based on `PSR-1 <https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-1-basic-coding-standard.md>`_
```
Comment (morozov) [inline docs/en/reference/index.rst]: Same as above, the full name would be cleaner.

File: README.md (hunk 3 above)
Comment (alcaeus) [inline README.md]: "The Doctrine Coding Standard is a set of rules for (link) PHP_CodeSniffer and applies to all Doctrine projects."

We could also add wording suggesting that people use it for their projects, although I'm unsure about the wording itself.

File: docs/en/reference/index.rst [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+and `PSR-2 <https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md>`_, with some noticeable
```
Comment (alcaeus) [inline docs/en/reference/index.rst]: I believe this section is redundant: PSR-2 itself is based on PSR-1, so I don't think we need to point this out.

File: README.md [hunk 6]
```diff
@@ -1,103 +1,13 @@
...
-
-If you are contributing to the Doctrine Coding Standard and want to test your contribution, you just
-need to execute PHPCS with the tests folder and ensure it matches the expected report:
-
-```bash
-$ ./vendor/bin/phpcs tests/input --report=summary --report-file=phpcs.log; diff tests/expected_report.txt phpcs.log
-```
+* [Website](https://www.doctrine-project.org/)
+* [Documentation](https://www.doctrine-project.org/projects/doctrine-coding-standard/en/latest/)
+* [Downloads](https://github.com/doctrine/persistence/coding-standard)
```
Comment (Majkl578) [inline README.md]: wrong link here: "persistence"

File: docs/en/index.rst [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+Getting Help
+------------
+
+If this documentation is not helping to answer questions you have about
+Doctrine Coding Standards don't panic. You can get help from different sources:
+
+-  The `Doctrine Mailing List <https://groups.google.com/group/doctrine-user>`_
+-  Gitter chat room `#doctrine/coding-standards <https://gitter.im/doctrine/coding-standards>`_
+-  Report a bug on `GitHub <https://github.com/doctrine/coding-standards/issues>`_.
```
Comment (Majkl578) [inline docs/en/index.rst]: wrong link here: "coding-standards" should be "coding-standard"

File: docs/en/index.rst [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+to know about the coding standards project.
+
+Getting Help
+------------
+
+If this documentation is not helping to answer questions you have about
+Doctrine Coding Standards don't panic. You can get help from different sources:
+
+-  The `Doctrine Mailing List <https://groups.google.com/group/doctrine-user>`_
+-  Gitter chat room `#doctrine/coding-standards <https://gitter.im/doctrine/coding-standards>`_
```
Comment (Majkl578) [inline docs/en/index.rst]: wrong link here: "coding-standards" should be "coding-standard"

File: docs/en/index.rst [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+-  Report a bug on `GitHub <https://github.com/doctrine/coding-standards/issues>`_.
+-  On `StackOverflow <https://stackoverflow.com/questions/tagged/doctrine-coding-standards>`_
```
Comment (Majkl578) [inline docs/en/index.rst]: wrong link here: "coding-standards" should be "coding-standard"

File: README.md [hunk 10]
```diff
@@ -1,103 +1,13 @@
-Doctrine Coding Standard
-========================
+# Doctrine Coding Standard
 
 [![Build Status](https://img.shields.io/travis/doctrine/coding-standard/master.svg?style=flat-square)](http://travis-ci.org/doctrine/coding-standard)
 [![Total Downloads](https://img.shields.io/packagist/dt/doctrine/coding-standard.svg?style=flat-square)](https://packagist.org/packages/doctrine/coding-standard)
 [![Latest Stable Version](https://img.shields.io/packagist/v/doctrine/coding-standard.svg?style=flat-square)](https://packagist.org/packages/doctrine/coding-standard)
 
+The Doctrine Coding Standard is a set of [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer) rules applied to all Doctrine projects.
```
Comment (Majkl578) [inline README.md]: Can we also give attribution here to the Slevomat guys / @kukulich for their amazing work on Slevomat CS? Something like: 
```markdown
Doctrine Coding Standard is heavily based on [Slevomat Coding Standard](https://github.com/slevomat/coding-standard).
```

**Pull request merged**